de-lib/src/main/java/org/iplantc/de/teams/client/presenter/TeamSearchRpcProxy.java

```diff
@@ -65,17 +72,66 @@ public void onFailure(Throwable caught) {
 
             @Override
             public void onSuccess(List<Group> result) {
-                PagingLoadResultBean<Group> loadResult = getLoadResult(result, loadConfig);
-                ensureHandlers().fireEvent(new TeamSearchResultLoad(result));
-                callback.onSuccess(loadResult);
+                getTeamCreatorNames(result, loadConfig, callback);
             }
         });
     }
 
+    void getTeamCreatorNames(List<Group> teams, FilterPagingLoadConfig loadConfig, AsyncCallback<PagingLoadResult<Group>> callback) {
+        if (teams != null && !teams.isEmpty()) {
+
+            List<String> subjectIds = getCreatorIds(teams);
+            collaboratorsFacade.getUserInfo(subjectIds, new AsyncCallback<FastMap<Subject>>() {
+                @Override
+                public void onFailure(Throwable throwable) {
+                    announcer.schedule(new ErrorAnnouncementConfig(appearance.getCreatorNamesFailed()));
+                    sendResults(teams, loadConfig, callback);
+                }
+
+                @Override
+                public void onSuccess(FastMap<Subject> creatorFastMap) {
+                    addCreatorToTeams(teams, creatorFastMap);
+                    sendResults(teams, loadConfig, callback);
+                }
+            });
+        } else {
+            sendResults(teams, loadConfig, callback);
+        }
+    }
+
+    void sendResults(List<Group> teams, FilterPagingLoadConfig loadConfig, AsyncCallback<PagingLoadResult<Group>> callback) {
+        PagingLoadResultBean<Group> loadResult = getLoadResult(teams, loadConfig);
+        ensureHandlers().fireEvent(new TeamSearchResultLoad(teams));
+        callback.onSuccess(loadResult);
+    }
+
     PagingLoadResultBean<Group> getLoadResult(List<Group> result, FilterPagingLoadConfig loadConfig) {
         return new PagingLoadResultBean<>(result, result.size(), loadConfig.getOffset());
     }
 
+    List<String> getCreatorIds(List<Group> teams) {
+        return teams.stream().map(this::getCreatorId).distinct().collect(Collectors.toList());
+    }
+
+    void addCreatorToTeams(List<Group> teams, FastMap<Subject> creatorFastMap) {
+        teams.forEach(group -> group.setCreator(creatorFastMap.get(getCreatorId(group)).getSubjectDisplayName()));
```

Comment:
Since `getCreatorId` has the possibility to return `null`, it seems like this line could throw an NPE (either by the `FastMap` or from the `getSubjectDisplayName` call).

Maybe we should check the return value of `getCreatorId` for null?